build/ci/official.yml

```diff
@@ -61,117 +88,182 @@ steps:
 - task: CopyFiles@2
+  displayName: Copy Symbols
   inputs:
     SourceFolder: '$(Build.SourcesDirectory)\artifacts\$(BuildConfiguration)'
     Contents: |
       **/Microsoft.VisualStudio.ProjectSystem.Managed?(*.pdb|*.dll|*.xml)
       **/Microsoft.VisualStudio.AppDesigner?(*.pdb|*.dll|*.xml)
       **/Microsoft.VisualStudio.Editors?(*.pdb|*.dll|*.xml)
-    TargetFolder: $(Build.ArtifactStagingDirectory)/symbols
-  displayName: Copy symbols
+    TargetFolder: $(Build.ArtifactStagingDirectory)/Symbols
 
+# https://docs.microsoft.com/azure/devops/pipelines/tasks/build/index-sources-publish-symbols
 - task: PublishSymbols@2
-  displayName: Archive symbols to VSTS
+  displayName: Publish Symbols to Azure Artifacts
   inputs:
-    SymbolsFolder: $(Build.ArtifactStagingDirectory)/symbols
-    SearchPattern: 
-      '**/*.pdb
+    SymbolsFolder: '$(Build.ArtifactStagingDirectory)/Symbols'
+    SearchPattern: |
+      **/*.pdb
       **/*.dll
-      **/*.exe'
-    publishSymbols: true
+      **/*.exe
+    # NOTE: This produces a warning about indexing (if enabled) but will still publish the symbols. See: https://github.com/microsoft/azure-pipelines-tasks/issues/15605
+    IndexSources: false
+    PublishSymbols: true
     SymbolServerType: TeamServices
-    SymbolsProduct: 'Roslyn Project System'
+    SymbolsProduct: 'DotNet Project System'
     SymbolsVersion: '$(Build.BuildNumber)'
     SymbolsArtifactName: Symbols
   continueOnError: true
 
+# TODO: Most of these files are already published via 'Publish Artifacts' section. Investigate publish consolidation.
 - task: PublishBuildArtifacts@1
+  displayName: 'Publish Artifact: Symbols'
   inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)/symbols'
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)/Symbols'
+    # The artifact name specifically has to be 'symbols' with that casing.
+    # This artifact is downloaded by the release pipeline's 'Publish Symbols' task.
+    # The lowercase 'symbols' is specified in the 'Agent job' settings for that task.
+    # If this artifact name is changed, all release pipelines that have this task need to have these settings changed.
     ArtifactName: symbols
     publishLocation: Container
-  displayName: 'Publish Artifact: symbols'
-        
+
+###################################################################################################################################################################
+# PUBLISH OPTPROF METADATA
+###################################################################################################################################################################
+
+# This is required to deploy the tests to devdiv.artifacts.visualstudio.com for the 'Deploy tests' step in the release pipeline.
+- task: artifactDropTask@0
+  displayName: 'Publish to Artifact Services: ProfilingInputs'
+  inputs:
+    dropServiceURI: 'https://devdiv.artifacts.visualstudio.com'
```

Comment:
Is the use of `visualstudio.com` correct here? I thought these URLs had been migrated to `azure.com`. That's certainly the case for AzDO boards and repos.

There are other usages of `visualstudio.com` elsewhere in this file too.

Reply:
I think you're referring that these go to the same place:
- https://devdiv.visualstudio.com/
- https://dev.azure.com/devdiv/

They just have a redirect service running for this stuff. I haven't seen a newer version of this endpoint, since it is the drop service endpoint and not a webpage endpoint. Just took a quick search in //index and the usage of this task, `artifactDropTask`, use this same endpoint. If there is an alternative, no one is using it. Got it from the docs here: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/1877/OptProf?anchor=build-definition-tasks